src/js/components/covid19/recipient/map/MapWrapper.jsx

```diff
@@ -112,11 +116,8 @@ export default class MapWrapper extends React.Component {
 
     mapReady = () => {
         // map has mounted, load the state shapes
-        this.setState({
-            mapReady: true
-        }, () => {
-            this.prepareMap();
-        });
+        this.props.onMapLoaded(true);
+        this.prepareMap();
```

Comment:
just a question without digging deeper: should these lines be reversed? will there be any timing or logic problems if onMapLoaded finishes before prepareMap starts?

Reply:
Thanks for asking this question. There were some pieces missing here. I added a condition to `componentDidUpdate` to replicate the behavior above.